### PR TITLE
noaa-apt: init at 1.2.0

### PR DESCRIPTION
**Pull request by trepetti**
###### Motivation for this change

[noaa-apt](https://noaa-apt.mbernardi.com.ar/) is decoding software that enables the conversion of audio recordings of NOAA weather satellite APT protocol transmissions into a variety of image formats. Typically this recording is done using software defined radios (SDRs) like RTLSDR, HackRF, etc., which are already well supported in Nix. This software would complement the growing suite of SDR tools.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/radio/noaa-apt/default.nix [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    atk
+    cairo
+    gdk-pixbuf
+    glib
+    gtk3
+    openssl
+    pango
+  ];
+
+  enableParallelBuilding = true;
```
Comment (danieldk) [inline pkgs/applications/radio/noaa-apt/default.nix:39]:
```suggestion
```

Not needed for `buildRustPackage`.

File: pkgs/applications/radio/noaa-apt/default.nix [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    mkdir -p $out/share/noaa-apt
+    cp -R $src/res/* $out/share/noaa-apt/
+
+    # Desktop icon.
+    mkdir -p $out/share/applications
+    mkdir -p $out/share/icons/hicolor/48x48/apps
+    mkdir -p $out/share/icons/hicolor/scalable/apps
+    cp $src/debian/ar.com.mbernardi.noaa-apt.desktop $out/share/applications/
+    cp $src/debian/noaa-apt.png $out/share/icons/hicolor/48x48/apps/
+    cp $src/debian/noaa-apt.svg $out/share/icons/hicolor/scalable/apps/
```
Comment (danieldk) [inline pkgs/applications/radio/noaa-apt/default.nix:59]: (Untested, but to give an idea.)

```suggestion
    install -Dm644 -t $out/share/applications $src/debian/ar.com.mbernardi.noaa-apt.desktop 
    install -Dm644 -t $out/share/icons/hicolor/48x48/apps $src/debian/noaa-apt.png 
    install -Dm644 -t $out/share/icons/hicolor/scalable/apps $src/debian/noaa-apt.svg 
```

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -4568,7 +4568,7 @@ in
 
   isync = callPackage ../tools/networking/isync { };
 
-  itm-tools = callPackage ../development/tools/misc/itm-tools { };  
+  itm-tools = callPackage ../development/tools/misc/itm-tools { };
```
Comment (danieldk) [inline pkgs/top-level/all-packages.nix:4571]: Unrelated changes shouldn't be in this commit.

File: pkgs/applications/radio/noaa-apt/default.nix [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  '';
+
+  meta = with lib; {
+    description = "NOAA APT image decoder.";
+    homepage = "http://noaa-apt.mbernardi.com.ar/";
+    license = licenses.gpl3;
```
Comment (danieldk) [inline pkgs/applications/radio/noaa-apt/default.nix:65]: `gpl3` is deprecated, use either `gpl3Only` or `gpl3Plus`. Can't find the 'or later wording anywhere, so we should probably assume `gpl3Only`:

```suggestion
    license = licenses.gpl3Only;
```

File: pkgs/applications/radio/noaa-apt/default.nix [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    maintainers = with maintainers; [ trepetti ];
+    platforms = platforms.linux ++ platforms.darwin;
```
Comment (danieldk) [inline pkgs/applications/radio/noaa-apt/default.nix:67]: Seems like this would work then for `platforms.all`, if so, you could simply remove this line.

```suggestion
```

File: pkgs/applications/radio/noaa-apt/default.nix [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+  };
```
Comment (danieldk) [inline pkgs/applications/radio/noaa-apt/default.nix:68]: Upstream does seem to do proper changelogs addressable by version, so this would be nice to add:

```suggestion
    changelog = "https://github.com/martinber/noaa-apt/releases/tag/v${version}";
  };
```

**Review by danieldk — CHANGES_REQUESTED** [review]
Thanks. I added a number of small requested changes.

File: pkgs/top-level/all-packages.nix (hunk 3 above)
Comment (trepetti) [reply pkgs/top-level/all-packages.nix:4571]: Thank you! My Vim configuration deletes trailing whitespace upon file write. Now fixed.

File: pkgs/applications/radio/noaa-apt/default.nix (hunk 4 above)
Comment (trepetti) [reply pkgs/applications/radio/noaa-apt/default.nix:65]: Done.

File: pkgs/applications/radio/noaa-apt/default.nix (hunk 5 above)
Comment (trepetti) [reply pkgs/applications/radio/noaa-apt/default.nix:67]: Sounds about right. I'm not very familiar with BSD and any other experimental platforms Nix supports, so I erred on the side of caution. All the libraries here are just GUI/graphics and should be cross-platform, though.

File: pkgs/applications/radio/noaa-apt/default.nix (hunk 6 above)
Comment (trepetti) [reply pkgs/applications/radio/noaa-apt/default.nix:68]: Didn't know about this feature. Thanks!

**Comment by trepetti** [comment]
Thank you for your review! I believe this covers all those fixes.

**Review by danieldk — APPROVED** [review]
LGTM, thanks for making all the changes!

Result of `nixpkgs-review pr 96519` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- noaa-apt
</details>
